pkg/kapp/diff/resource_with_history.go

```diff
@@ -218,5 +211,11 @@ func (resourceWithoutHistory) removeAppliedResAnnKeysMods() []ctlres.ResourceMod
 			ResourceMatcher: ctlres.AllMatcher{},
 			Path:            ctlres.NewPathFromStrings([]string{"metadata", "annotations", debugAppliedResDiffFullAnnKey}),
 		},
+		// metadata.managedFields technically is not kapp history, but we want to strip it in same situations we are
+		// stripping kapp history, it is server side apply history after all
+		ctlres.FieldRemoveMod{
```

Comment:
without this, was this breaking anything? (if so, just for ssa, or for non-ssa operation?)

Reply:
Yes, with SSA enabled it was reporting diff on `managedFields` where it shouldn't